File: samples/client/petstore/java/src/test/java/org/openapijsonschematools/schemas/validation/JsonSchemaTest.java
package org.openapijsonschematools.schemas.validation;

import org.junit.Assert;
import org.junit.Test;
import org.openapijsonschematools.configurations.JsonSchemaKeywordFlags;
import org.openapijsonschematools.configurations.SchemaConfiguration;

import java.util.ArrayList;
import java.util.LinkedHashMap;
import java.util.LinkedHashSet;
import java.util.List;
import java.util.Set;

class SomeSchema extends JsonSchema {
    static final LinkedHashSet<Class<?>> type = new LinkedHashSet<>(Set.of(
        String.class
    ));
}

public class JsonSchemaTest {

    @Test
    public void testValidateSucceeds() {
        List<Object> pathToItem = new ArrayList<>();
        pathToItem.add("args[0");
        ValidationMetadata validationMetadata = new ValidationMetadata(
                pathToItem,
                new SchemaConfiguration(JsonSchemaKeywordFlags.ofNone()),
                new PathToSchemasMap(),
                new LinkedHashSet<>()
        );
        PathToSchemasMap pathToSchemas = JsonSchema.validate(
                SomeSchema.class,
                "hi",
                validationMetadata
        );
        PathToSchemasMap expectedPathToSchemas = new PathToSchemasMap();
        LinkedHashMap<Class<? extends JsonSchema>, Void> validatedClasses = new LinkedHashMap<>();
        validatedClasses.put(SomeSchema.class, null);
        expectedPathToSchemas.put(pathToItem, validatedClasses);
        Assert.assertEquals(pathToSchemas, expectedPathToSchemas);
    }
}

File: samples/client/3_1_0_unit_test/kotlin/src/test/kotlin/org/openapijsonschematools/client/schemas/validation/JsonSchemaTest.kt
package org.openapijsonschematools.client.schemas.validation

import org.openapijsonschematools.client.configurations.SchemaConfiguration
import org.openapijsonschematools.client.exceptions.ValidationException

import kotlin.test.Test
import kotlin.test.assertEquals

class JsonSchemaTest {
    internal sealed interface SomeSchemaBoxed
    internal class SomeSchemaBoxedString : SomeSchemaBoxed
    internal class SomeSchema private constructor() : JsonSchema<SomeSchemaBoxed>(
        type = setOf(String::class.java)
    ) {
        override fun getNewInstance(arg: Any?, pathToItem: List<Any>, pathToSchemas: PathToSchemasMap): Any? {
            if (arg is String) {
                return arg
            }
            throw RuntimeException("Invalid input type=$javaClass. It can't be instantiated by this schema")
        }

        @Throws(ValidationException::class)
        override fun validate(arg: Any?, configuration: SchemaConfiguration?): Any? {
            if (arg is String) {
                return arg
            }
            throw ValidationException("Invalid input type=$javaClass. It can't be validated by this schema")
        }

        @Throws(ValidationException::class)
        override fun validateAndBox(arg: Any?, configuration: SchemaConfiguration?): SomeSchemaBoxed {
            return SomeSchemaBoxedString()
        }

        companion object {
            var instance: SomeSchema? = null
                get() {
                    if (field == null) {
                        field = SomeSchema()
                    }
                    return field
                }
                private set
        }
    }

    @Test
    @Throws(ValidationException::class)
    fun testValidateSucceeds() {
        val pathToItem = listOf<Any>("args[0]")
        val validationMetadata = ValidationMetadata(
            pathToItem,
            SchemaConfiguration(),
            PathToSchemasMap(),
            LinkedHashSet()
        )
        val schema = JsonSchemaFactory.getInstance(SomeSchema::class.java)
        val pathToSchemas = JsonSchema.validate(
            schema,
            "hi",
            validationMetadata
        )
        val expectedPathToSchemas = PathToSchemasMap()
        val validatedClasses = LinkedHashMap<JsonSchema<*>, Nothing?>()
        validatedClasses[schema] = null
        expectedPathToSchemas[pathToItem] = validatedClasses
        assertEquals(pathToSchemas, expectedPathToSchemas)
    }
}